### ISSUE TITLE
FID on Class-Conditioned Evaluation & Normalized Attn in Depth 16/30

### ISSUE THREAD

**Issue by lxa9867** [FoundationVision/VAR#38]
Hi Keyu,

Thanks for sharing this interesting work. I have two questions and hope to get some insights from you.

(1) I am trying to evaluate the released ckpt on the ImageNet eval set. The reproduced FID is about ~7 for the depth-16 model. I consider there should be a config difference between our testing (we used the configs in the released demo and torchmetrics to evaluate) and the reported experiment. Would you please let us know the specific parameters setting for sampling during the evaluation? Thanks.

(2) I noticed that in a recent update, the released ckpts were replaced and so were the configs of depth 16/30 var configs. I wonder what is the influence of normalized attention and what about the performance improvement based on this modification. Thanks.

Best,
A loyal reader of your paper

**Comment by keyu-tian** [comment]
hi @lxa9867 you can use the configurations at https://github.com/FoundationVision/VAR?tab=readme-ov-file#sampling--zero-shot-inference to sample 50k images (50 images per class). To reproduce FID, it needs to use the OpenAI evaluation protocol in https://github.com/openai/guided-diffusion/tree/main/evaluations.


Normalized attention can stablize the training when fp16 is used. If trained with fp32, it should yield similar results.

**Comment by lxa9867** [comment]
Got it. Thanks for the quick response!

**Comment by JamesHujy** [comment]
I have the similar question. In ADM, the fid is evaluated using the training set as the reference batch. Following this, I run the inference with depth-16 checkpoint and evaluate the FID score with stats on the training set provided by OpenAI. The result is ~10. It seems a config gap during my test.
